examples/end-to-end-examples/support_chatbot/README.md

```diff
@@ -0,0 +1,179 @@
+
+This example will contain a script for a [text to MDS](./scripts/conversion/convert_txt_to_stream.py) conversion. For text to MDS conversion, we will run:
+
+```bash
+mcli run -f mcli_yamls/conversion/convert_txt_to_stream.yaml --cluster CLUSTER
+```
+
+**The conversion YAMLs will not work on local directories if the data is not git pushed onto the repository linked by the YAML.** This means that it will not be able to recognize and oci path to data, but only a local path in the github repository (you can feed it the relative path from *step 1*). Thus, if you choose to use the YAML method, make sure that you push the data downloaded from *step* one to your repository. That being said, github repositories are typically very small and you can probably just run this locally:
+
+```bash
+python scripts/conversion/convert_txt_to_stream.py \
+    --out_root CLOUD://BUCKET/support-bot-demo/data/DATA_NAME_MDS/ \
+    --in_root PATH_TO_TXT_FOLDER
+```
+
+**Fields to replace with your values:** `CLUSTER` (in the command line), `CLOUD` (in the yaml), `BUCKET` (in the yaml), 'DATA_NAME_MDS' (in the yaml), `PATH_TO_TXT_FOLDER` (in the yaml). Please note that PATH_TO_TXT_FOLDER **MUST** be a local directory (not an OCI link) due to some limitations from OCI. To follow with our previous example in Step 1, if we want to convert the folder containing all of composer as text, we will run with `DATA_NAME_MDS = composer_codebase_mds` and `PATH_TO_TXT_FOLDER = retrieval_data/composer`
+
+## Step 3: Finetuning on our Repository
+
+Next, we will finetune our pretrained model on the train split of our data for the end task of answering questions about the codebase.
+
+Please check out the [training directory](./mcli-yamls/finetune) for all of the details. This yaml will load the pretrained weights for `mpt-30b` available on the [HuggingFace Hub](https://huggingface.co/mosaicml/mpt-30b), and then train using the normal causal language modeling objective on our datasets that we processed in the previous step. The [training script](https://github.com/mosaicml/llm-foundry/blob/main/scripts/train/train.py) itself, is from LLM-foundry.
+
+To run finetuning on the composer codebase, run the following:
+
+```bash
+mcli run -f mcli_yamls/finetune/finetune_composer_codebase.yaml --cluster CLUSTER
+```
+**Fields to replace with your values:** `CLUSTER` (in the command line), `CLOUD` (in the yaml), `BUCKET_NAME` (in the yaml).
+
+**Inputs:** `composer_codebase_mds` cloud folder from step 2
+
+**Outputs:** the checkpoints from your training, saved to the `save_folder` specified in the yaml
+
+## Step 4: Finetuning on Chatv2
+
+Next, we will finetune our model on the train split of the chatv2 dataset to ensure that the bot still has its QA abilities. Chatv2 is the dataset that was used to finetune the MPT-30B model to be MPT-30B-Chat. To run finetuning on Chatv2, run the following:
```

Comment:
lets add a note here that the chatv2 dataset includes GPT generated output, and link to `mosaicml/mpt-30b-chat` huggingface page for more info